src/core/frames/frame_controller.js

```diff
@@ -234,6 +233,9 @@ export class FrameController {
     const frame = this.#findFrameElement(formSubmission.formElement, formSubmission.submitter)
 
     frame.delegate.proposeVisitIfNavigatedWithAction(frame, getVisitAction(formSubmission.submitter, formSubmission.formElement, frame))
+    if (frame.src === response.response.url && frame.shouldReloadWithMorph) {
```

Comment:
Determining whether a page-wide Visit is to be handled as a refresh [involves more than an exact comparison of URL values](https://github.com/hotwired/turbo/blob/v8.0.10/src/core/drive/page_view.js#L58-L60).

At the Page level, it involves comparing the [URL.pathname](https://developer.mozilla.org/en-US/docs/Web/API/URL/pathname) values (ignoring differences in query parameters) *as well as* checking for whether or not the `Visit.action` is `"replace"`.

This change could make a similar comparison at the `[src]` and `URL` level, but driving Frame elements does not have a concept of a `Visit`, and `[data-turbo-action="replace"]` has Page-wide navigation implications.

Does it make sense to compare pathnames and ignore query parameters?

Reply:
@seanpdoyle Thanks for taking a look at this, and for pointing out how page refreshes already handle this situation regarding query params. I definitely want to try to keep things consistent between the frame and page systems! Not to mention that it will make this feature just that little bit more useful. I'll push a commit for this shortly.

Regarding #430, I love it. Initially wrapping my head around Turbo's internals has not been easy, and that additional abstraction would help to compartmentalize and clarify some of the complexity. I hope it finally gets merged soon.